lighthouse-core/fraggle-rock/gather/navigation-runner.js

```diff
@@ -327,6 +327,7 @@ async function navigationGather(requestor, options) {
   const artifacts = await Runner.gather(
     async () => {
       let {page} = options;
+      const normalizedRequestor = isCallback ? requestor : URL.normalizeUrl(requestor);
```

Comment:
hmm, is this change necessary? it isn't even used until far down.

Reply:
Yes, at least to match the legacy behavior. If the url is invalid, we should throw before attempting to connect to the page.